heron/io/dlog/src/java/org/apache/heron/dlog/DLInputStream.java

```diff
@@ -120,7 +99,7 @@ public int read(byte[] b, int off, int len) throws IOException {
     if (currentLogRecord == null) {
       currentLogRecord = nextLogRecord();
       if (currentLogRecord == null) {
-        return read;
+        return -1;
```

Comment:
The original logic seems to return a 0 and then return -1 in the next run if it is EOS. It seems to be safer to me (if currently there is no data but it is not EOS yet). Is there any issue?

Reply:
Yes, the protocol buffer message parser insists on not seeing any 0 return value, therefore the first 0 already results in an error. We have to return -1 immediately when the end is reached. I tested this on my Heron+Bookkeeper on k8s cluster with a stateful topology.

Reply:
kk. thx